File: VB/CustomData/Data/WebApiLookup.vb
Imports DevExtreme.AspNet.Data.ResponseModel
Imports Microsoft.AspNetCore.Components
Imports System.Text.Json
Imports DevExtreme.AspNet.Data
Imports DevExpress.Blazor

Namespace CustomData.Data

    Public Class WebApiLookup

        Public Property Text As String

        Public Property Value As String
    End Class

    Public Module CustomDataHelper

        Public Async Function LoadCustomData(ByVal client As HttpClient, ByVal options As DataSourceLoadOptionsBase, ByVal cancellationToken As CancellationToken) As Task(Of LoadResult)
            Dim response = Await client.GetAsync(options.ConvertToGetRequestUri("https://js.devexpress.com/Demos/NetCore/api/DataGridWebApi/CustomersLookup"), cancellationToken)
            response.EnsureSuccessStatusCode()
            Dim responseStream = Await response.Content.ReadAsStreamAsync()
            Return Await JsonSerializer.DeserializeAsync(Of LoadResult)(responseStream, cancellationToken:=cancellationToken)
        End Function
    End Module
End Namespace

File: CS/CustomData/Data/WebApiLookup.cs
using DevExtreme.AspNet.Data.ResponseModel;
using Microsoft.AspNetCore.Components;
using System.Text.Json;
using DevExtreme.AspNet.Data;
using DevExpress.Blazor;

namespace CustomData.Data {
    public class WebApiLookup {
        public string Text { get; set; }
        public string Value { get; set; }
    }

    public static class CustomDataHelper {
        public static async Task<LoadResult> LoadCustomData(HttpClient client, DataSourceLoadOptionsBase options, CancellationToken cancellationToken) {
            using var response = await client.GetAsync(options.ConvertToGetRequestUri
                ("https://js.devexpress.com/Demos/NetCore/api/DataGridWebApi/CustomersLookup"), cancellationToken);
            response.EnsureSuccessStatusCode();
            using var responseStream = await response.Content.ReadAsStreamAsync();
            return await JsonSerializer.DeserializeAsync<LoadResult>(responseStream, cancellationToken: cancellationToken);
        }
    }
}
